docs/proposals/gpu-support.md

```diff
@@ -0,0 +1,320 @@
+but, instead, it will be used to return the resources used by the pod when it
+terminates.
+
+Nodes with available resources are handed out in a circular fashion in order to
+
+- approximate round robin scheduling
+- increase the chances of forward progress when the same pod is passed to the
+  extender multiple times (e.g. because of other constraints not being met)
+
+### The runtime
+
+Once we know where to run the container, it's time to set up its environment. At
+a minimum, we'll need to map the host device(s) into the container. Because each
+manufacturer exposes different device nodes (`/dev/ati/card0`, `/dev/nvidia0`,
+but also the required `/dev/nvidiactl` and `/dev/nvidia-uvm`), some of the logic
+needs to be hardware-specific, mapping from a logical device to a list of device
+nodes necessary for software to talk to it.
+
+Support binaries and libraries are often versioned along with the kernel module,
+so there should be further hooks to project those under `/bin` and some kind of
+`/lib` before the application is started. This can be done for Docker with the
+use of a versioned [Docker
+volume](https://docs.docker.com/engine/userguide/containers/dockervolumes/). The
+alternative would be for each application to bundle everything in their image.
+
+#### NVIDIA support
+
+The first implementation and proof of concept will be for NVIDIA devices, the
+setup that is most common and with more extensive infrastructure, namely the
+[nvidia-docker plugin](https://github.com/NVIDIA/nvidia-docker). The plugin is
+an officially supported solution, thus avoiding a lot of new low level code, as
+it takes care of functionality such as:
+
+- creating a Docker volume with binaries such as `nvidia-smi` and shared
+libraries
+- providing HTTP endpoints that monitoring tools can use to collect GPU metrics
+- abstracting details such as `/dev` entry names for each device, as well as
+control ones like `nvidiactl`
+
+The `nvidia-docker` wrapper also verifies that the CUDA version required by a
+given image is supported by the host drivers, through inspection of well-known
+image labels, if present. We should try to provide equivalent checks, either
+for CUDA or OpenCL.
+
+This is current sample output from `nvidia-docker-plugin`, wrapped for
+readability:
+
+    $ curl -s localhost:3476/docker/cli
+    --device=/dev/nvidiactl --device=/dev/nvidia-uvm --device=/dev/nvidia0
+    --volume-driver=nvidia-docker
+    --volume=nvidia_driver_352.68:/usr/local/nvidia:ro
+
+It runs as a daemon listening for HTTP requests on port 3476. The endpoint above
+returns flags that need to be added to the Docker command line in order to
+expose GPUs to the containers. There are optional URL arguments to request
+specific devices if more than one are present on the system, as well as specific
+versions of the support software.
+
+An obvious improvement is an additional endpoint for JSON output. The unresolved
```

Comment:
Please remove the first line "An obvious improvement...". Or put it to previous paragraph.
It's really confusing here